File: Daily_Codes_Answers/day-8/136_Single_Number.py
def singleNumber(nums):
    if(len(nums)==1):
        return nums[0]
    
    nums.sort()
    for i in range(0,len(nums)-1,2):
        if(nums[i]!=nums[i+1]):
                return nums[i]
    return nums[len(nums)-1]
            
test_case_1 = [2,2,1]
test_case_2 = [4,1,2,1,2]
test_case_3 = [1]

print("-- test case 1 --")
print(f"Input : {test_case_1}\nOutput : {singleNumber(test_case_1)}")
print("-- test case 2 --")
print(f"Input : {test_case_2}\nOutput : {singleNumber(test_case_2)}")
print("-- test case 3 --")
print(f"Input : {test_case_3}\nOutput : {singleNumber(test_case_3)}")

File: Daily_Codes_Answers/day-8/136_Single_Number.js
const singleNumber = function(nums){
    if(nums.length==1){
        return nums[0];
    }
    nums.sort()
    for(let i=0 ; i<nums.length-1 ; i+=2){
        if(nums[i]!=nums[i+1]){
            return nums[i];
        }
    }
    return nums[nums.length-1]; 
}

let test_case_1 = [2,2,1]
let test_case_2 = [4,1,2,1,2]
let test_case_3 = [1]

console.log("-- test case 1 --")
console.log("Input : "+test_case_1+"\nOutput : "+singleNumber(test_case_1))
console.log("-- test case 2 --")
console.log("Input : "+test_case_2+"\nOutput : "+singleNumber(test_case_2))
console.log("-- test case 3 --")
console.log("Input : "+test_case_3+"\nOutput : "+singleNumber(test_case_3))